Ebs 146 - Global Styles To TS

sblaettgen opened the pull request:
### Refactoring Global Styles
Global Styles are converted to Typescript.

Since ShadowDom inherited from it's parent only a fraction of CSS Attributes the use of wrapper around the WebComponents is needed to use global styling. It is needed to break up the Global Styles to use them inside the extra WrapperStyle. It should keep it DRY and the wrapper itself receive it's content from a SSOT so to speak - Global Styles stay in the global style files.

derblaettgen committed:
renamed js file to ts

derblaettgen committed:
renamed fonts js files to ts

derblaettgen committed:
added a declaration so TS compiles and knows what to do with fonts

derblaettgen committed:
Splitted GlobalStyling To Use It In ShadowStyling

derblaettgen committed:
Changed ReadMe, Yarn Uses Add

derblaettgen committed:
Font Faces Must Be Loaded Outside Of Shadow Dom

sblaettgen commented on src/GlobalStyles/global.style.ts:
```diff
@@ -1,13 +1,15 @@
-import { createGlobalStyle } from "styled-components";
```
It looks like I used squashed into the wrong direction, `global.style.ts `and `index.ts` shouldn't be behind that commit message.

derblaettgen committed:
Deleted Theme

Fixed Linter

Weetbix reviewed (approved):
Looks good!

Thanks for the changes :)

Weetbix commented on src/GlobalStyles/global.style.ts:
```diff
@@ -0,0 +1,28 @@
+import { createGlobalStyle, css } from "styled-components";
+import { Colors } from "..";
```
Update to the full import path as discussed :) 

like sidebar:
`import Colors from "../Colors";`

derblaettgen committed:
Updated Styled Components To Match Portal Version

derblaettgen committed:
Changed To Prevent Deprecated Imports

derblaettgen committed:
Seperate Everything To Prevent Webpack Witchcraft

derblaettgen committed:
Topmost Div Should Fill Its Layout

derblaettgen committed:
Updated Peer Dependencies

derblaettgen committed:
revert version change && add build
styled-components v5 seems to cause some errors in the storybook tests
FE decided to add the storybook build to the test script.

derblaettgen committed:
Updated Readme - Contributing Section

a7madgamal commented on src/Fonts/ShadowStyleFonts.ts:
```diff
@@ -0,0 +1,15 @@
+// added to the Shadow Styles
+const ShadowStylesFonts = css`
+  input,
+  textarea {
```
does this affect input and textarea only?

sblaettgen replied:
Nice that you adressing this ... yes indeed.

It is a straight convert from the original file but it doesn't make sense for me either. It is the default font, which is also used in `body` ... therefore I would suggest to get rid of `input, textarea` and introduce him as default standard for everything.

a7madgamal commented on src/Fonts/fonts.style.ts:
```diff
@@ -0,0 +1,120 @@
+import DefaultFont from "./DefaultFont";
+import ShadowStylesFonts from "./ShadowStyleFonts";
+
+import ie9Compat00 from "../../public/fonts/TTNorms/35C396_0_0.eot";
```
Nitpick, but it looks like lots of legacy stuff I hope we can get rid of some

sblaettgen replied:
Yes, I just converted it and get it to work with the ShadowDOM. Also I wasn't aware at that moment that the library is full of deprecated legacy code. Maybe it is a job for someone who know's whats been used and what not to get rid of it.

a7madgamal reviewed (approved):
nice job, if you tested it on the other repo and it works as expected then lgtm :)

a7madgamal commented on README.md:
```diff
@@ -27,7 +27,9 @@ focused in scope while avoiding unrelated commits and changes.
 - Add a reviewer (or ask someone in person) to go over your changes
-- If all goes well click the "Squash and Merge" button to close your PR (make sure to clean up and delete your branch)
+- When the PR is approved click the "Squash and Merge" button to close your PR (make sure to clean up and delete your branch)
+- Create a new release by following previous releases format and pick a relevant version according to the change (patch, minor, major)
```
maybe mention github clearly or even better add this link https://github.com/vivyteam/vivy-components/releases

a7madgamal commented on README.md:
```diff
@@ -27,7 +27,9 @@ focused in scope while avoiding unrelated commits and changes.
-- If all goes well click the "Squash and Merge" button to close your PR (make sure to clean up and delete your branch)
+- When the PR is approved click the "Squash and Merge" button to close your PR (make sure to clean up and delete your branch)
+- Create a new release by following previous releases format and pick a relevant version according to the change (patch, minor, major)
+- If all goes well you should receive an email from npm that the publish went ok
```
also here you can add this link for debugging if the job failed for some reason https://app.circleci.com/pipelines/github/VivyTeam/vivy-components?branch=master

derblaettgen committed:
Added Pipeline Link In Readme

Weetbix reviewed (approved):
Great thanks!
